server/neptune/workflows/internal/deploy/terraform/runner.go

```diff
@@ -78,8 +91,17 @@ func (r *WorkflowRunner) awaitWorkflow(ctx workflow.Context, future workflow.Chi
 		}
 	}
 
-	if err != nil {
-		return errors.Wrap(err, "executing terraform workflow")
+	// if we have an app error we should attempt to unwrap it's details into our own
+	// application error and act accordingly
+	var appErr *temporal.ApplicationError
+	if errors.As(err, &appErr) {
```

Comment:
do we also need to handle `UpdateJobError` here?

Reply:
no its a retryable error type.